### PR TITLE
add example FIT badge, formalize pat in doc and style option

### LINKED ISSUE #85: [Feature]: add option to pass in a token to style and doc workflow
### Is your feature request related to a problem? Please describe.

For [FIMS](https://github.com/NOAA-FIMS/FIMS), running the call-r-cmd-check workflow is required, but it will not be triggered with `GITHUB_TOKEN` (see https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow). However, passing in a PAT or using a GitHub app would allow opening a PR to trigger the workflow.

### Describe the solution you would like.

Allow users of syle and document to optionally pass in a PAT to the workflow. 

### Describe alternatives you have considered

Turn off the required check for FIMS and continue using call-r-cmd-check as normal.

### Additional context

see https://github.com/NOAA-FIMS/FIMS/pull/414

### PR DESCRIPTION
**Pull request by k-doering-NOAA**

### CODE REVIEW

**Comment by k-doering-NOAA** [comment]
@Bai-Li-NOAA, do you think you would be able to review this PR within the next week or so?

File: tests/testthat/test-use_r_workflows.R [hunk 1]
```diff
@@ -73,6 +73,26 @@ test_that("use_doc_and_style_r() works with manual trigger", {
...
 })
 
+test_that("use_doc_and_style_r() works with pat option", {
+  use_doc_and_style_r(
+    workflow_name = "doc_style_pat.yml",
+    use_rm_dollar_sign = FALSE, how_to_commit = "pull_request", 
+    use_pat = TRUE, pat_name = "MYPAT"
+  )
+  expect_true(file.exists(".github/workflows/doc_style_pat.yml"))
+  test <- readLines(".github/workflows/doc_style_pat.yml")
```
Comment (Bai-Li-NOAA) [inline tests/testthat/test-use_r_workflows.R:83]: comment [non-blocking]: maybe testing output of the R functions with [snapshot tests](https://testthat.r-lib.org/articles/snapshotting.html)

**Review by Bai-Li-NOAA — APPROVED** [review]
Looks great! Excited for the release!

File: tests/testthat/test-use_r_workflows.R (hunk 1 above)
Comment (k-doering-NOAA) [reply tests/testthat/test-use_r_workflows.R:83]: great point, I just learned about them from the ones (I assume) you wrote for FIMS!

**Pull request merged**